src/tools/tests/storage-key-recipe-resolver-test.ts

```diff
@@ -312,6 +312,27 @@ describe('recipe2plan', () => {
         assert.isTrue(it.isResolved());
       }
     });
+    it('forces resolution of type variables', async () => {
+      const manifest = await Manifest.parse(`\
+    schema Thing 
+      name: Text
+     
+    particle Writer
+       data: writes [Thing {name}] // Could also be a singleton
+    
+    @trigger
+      launch startup
+      arcId writeArcId
+    recipe WritingRecipe
+      thing: create persistent 'my-handle-id' 
+      Writer
+        data: thing`);
+
+      const resolver = new StorageKeyRecipeResolver(manifest);
+      for (const it of (await resolver.resolve())) {
+        assert.isTrue(it.isResolved());
```

Comment:
could you also verify that the type in the recipe can be resolved?

Reply:
doesn't recipe.isResolved() check that?

Reply:
actually should even assert that `type.getEntitySchema()` isn't null, which is the error that Satakshi was experiencing (but the recipe was resolved).